File: time-management-advisor/src/main/java/com/sbnz/timemanagementadvisor/model/Activity.java
package com.sbnz.timemanagementadvisor.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.sbnz.timemanagementadvisor.model.enums.ActivityContinuityType;
import com.sbnz.timemanagementadvisor.model.enums.ActivityType;
import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.Id;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Null;
import java.time.DayOfWeek;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@Document
public class Activity {
    @Id @JsonSerialize(using = ToStringSerializer.class)
    private ObjectId id;
    private String title, description;
    private List<CheckListItem> checklist;
    @NotNull
    private LocalDateTime start;
    private LocalDateTime end;
    private Long estimatedTimeMinutes;
    private Integer priority;
    @NotNull
    private ActivityContinuityType continuityType;
    @NotNull
    private ActivityType activityType;
    private Boolean isArchived=false, isDone=false;

    private Set<DayOfWeek> forDays; // only if periodic

}

File: frontend/time-management-app/src/model/Activity.ts
import { CheckListItem } from "./CheckListItem";
import { ActivityContinuityType } from "./enums/ActivityContinuityType";
import { ActivityType } from "./enums/ActivityType";
import { DayOfWeek } from "./enums/DayOfWeek";

export interface Activity {
  id: string;
  title: string;
  description: string;
  checklist: CheckListItem[];
  start: Date;
  end: Date;
  estimatedTimeMinutes: number;
  priority: number;
  continuityType: ActivityContinuityType;
  activityType: ActivityType;
  isArchived: boolean;
  isDone: boolean;
  forDays: DayOfWeek[];
}
